gapic/src/main/com/google/gapid/perfetto/models/FrameEventsTrack.java

```diff
@@ -253,60 +180,53 @@ public Data(DataRequest request, long[] ids, long[] starts, long[] ends,
   }
 
   public static class Slices implements Selection<Slices> {
-    private int count = 0;
+    public int count = 0;
     public final List<Long> ids = Lists.newArrayList();
     public final List<Long> times = Lists.newArrayList();
     public final List<Long> durs = Lists.newArrayList();
     public final List<String> names = Lists.newArrayList();
     public final List<ArgSet> argsets = Lists.newArrayList();
-    public final List<Long[]> frameNumbers = Lists.newArrayList();
-    public final List<String[]> layerNames = Lists.newArrayList();
-    // Map of each buffer(layerName) that contributed to the displayed frame, to
-    // its corresponding FrameStats
-    public final List<Map<String, FrameStats>> frameStats = Lists.newArrayList();
-    public final FrameSelection frameSelection = new FrameSelection();
+    public final List<Long> frameNumbers = Lists.newArrayList();
+    public final List<String> layerNames = Lists.newArrayList();
+    public final List<String> eventNames = Lists.newArrayList();
+    public final List<Long> queueToAcquireTimes = Lists.newArrayList();
+    public final List<Long> acquireToLatchTimes = Lists.newArrayList();
+    public final List<Long> latchToPresentTimes = Lists.newArrayList();
     public final Set<Long> sliceKeys = Sets.newHashSet();
+    public final Set<Long> selectedFrameNumbers = Sets.newHashSet();
 
-    public Slices(QueryEngine.Row row, ArgSet args, Map<String, FrameStats> frameStats) {
+    public Slices(QueryEngine.Row row, ArgSet args, String layerName, String eventName) {
       add(row.getLong(0), row.getLong(1), row.getLong(2), row.getString(3), args,
-          row.getString(8), row.getString(9), frameStats);
+          row.getLong(8), layerName, eventName, row.getLong(9), row.getLong(10), row.getLong(11));
     }
 
-    public Slices(QueryEngine.Row row, Map<String, FrameStats> frameStats) {
+    public Slices(QueryEngine.Row row, String layerName, String eventName) {
       add(row.getLong(0), row.getLong(1), row.getLong(2), row.getString(3), ArgSet.EMPTY,
-          row.getString(8), row.getString(9), frameStats);
+          row.getLong(8), layerName, eventName, row.getLong(9), row.getLong(10), row.getLong(11));
```

Comment:
So I removed the second method since it was only being used previously.  The method below this is actually used for an entire query result, rather than a row. Technically, from the caller's perspective, there is no args that can be applied for multiple rows. So, the ArgSet.EMPTY should be applied in the constructor rather than sending it from the caller.